internal/prometheusbp/spectest/spec_test.go

```diff
@@ -0,0 +1,326 @@
+package spectest
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/reddit/baseplate.go/errorsbp"
+)
+
+func TestMissingMetrics(t *testing.T) {
+	testCases := []struct {
+		name    string
+		missing map[string]struct{}
+		want    []string
+	}{
+		{
+			name:    "none missing",
+			missing: map[string]struct{}{},
+			want:    []string{},
+		},
+		{
+			name:    "missing",
+			missing: map[string]struct{}{"foo": {}},
+			want:    []string{"foo"},
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			if got, want := keysFrom(tt.missing), tt.want; !reflect.DeepEqual(got, want) {
```

Comment:
nit: instead of using `reflect.DeepEqual`, I would prefer to write a helper function to compare string slices:

```
func compareStringSlices(tb testing.TB, got, want []string) {
  ...
}
```

because I believe `reflect.DeepEqual([]string{}, []string(nil))` would return false, but they are both correct (we don't want to fail the tests when in the future the functions change to return `nil` for empty slice).

see: https://go.dev/play/p/02-mlRXai9O